### PR TITLE
feature(h2c): optional certificates for TLS

### PR DESCRIPTION
**Pull request by cv65kr**
# Reason for This PR

In case of h2c there a no need to provide certificates with enabled TLS option

## Description of Changes

If TLS is enabled certificates files are optional.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

**Comment by rustatian** [comment]
Hey @cv65kr 👋 
Thank you for the PR 👍
I see an error in the CI: https://github.com/temporalio/roadrunner-temporal/actions/runs/11846144712/job/33013137624?pr=579#step:15:9696

File: tls.go [hunk 1]
```diff
@@ -43,9 +43,8 @@ func initTLS(cfg *Config) (*tls.Config, error) {
 	}
 
 	return &tls.Config{
-		ServerName:           cfg.TLS.ServerName,
-		MinVersion:           tls.VersionTLS12,
-		GetClientCertificate: getClientCertificate(cfg),
```
Comment (rustatian) [inline tls.go:48]: Unfortunately, we can't remove `getClientCertificate`. It is used to hot-reload certificates.

**Comment by rustatian** [comment]
I guess it is better to introduce an option in the TLS config, like, `use_h2c: true/false`. And if used, skip certificates checking and skip `getClientCertificates` method (just return nil based on the config).

**Comment by cv65kr** [comment]
@rustatian changed

**Review by rustatian — APPROVED** [review]

**Comment by rustatian** [comment]
Thank you, @cv65kr 👍 
I have a plan to release 2024.3 with that change this Thursday.

**Pull request merged**